SYCL/ESIMD/api/functional/ctors/common.hpp

```diff
@@ -0,0 +1,165 @@
+//===-- common.hpp - This file provides common functions for simd constructors
+//      tests -------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// Common file for test on simd constructors.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl.hpp>
+#include <sycl/ext/intel/experimental/esimd.hpp>
+
+#include "../../../esimd_test_utils.hpp"
+#include "../logger.hpp"
+#include "../type_coverage.hpp"
+
+#include <type_traits>
+
+namespace esimd_test {
+namespace api {
+namespace functional {
+
+// Dummy kernel for submitting some code into device side.
+template <typename DataT, int NumElems, typename T> struct Kernel;
+
+template <typename DataT>
+using allocator_type = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
+
+template <typename DataT>
+using vector_with_allocator = std::vector<DataT, allocator_type<DataT>>;
```

Comment:
```suggestion
// Vector type to share data between device and host sides using USM allocator
template <typename DataT>
using shared_vector = std::vector<DataT, allocator_type<DataT>>;
```
Just as a suggestion

Reply:
I've done it in this commit: 573abdb97eeea16c695266a7f80c965d0ec8f989

Reply:
Please, feel free to close the resolved ones. Looks like I cannot do it myself for some reason.